src/entities/liquidity-mining-campaign.ts

```diff
@@ -0,0 +1,111 @@
+import { BigintIsh, ChainId, SECONDS_IN_YEAR } from '../constants'
+import JSBI from 'jsbi'
+import { parseBigintIsh } from '../utils'
+import { CurrencyAmount, Fraction, Percent } from './fractions'
+import { PricedTokenAmount } from './fractions/priced-token-amount'
+import { Pair } from './pair'
+import invariant from 'tiny-invariant'
+import { utils } from 'ethers'
+import { Token } from './token'
+
+// this value is used as a floor to calculate apy, in order to avoid infinite results
+const MINIMUM_STAKED_AMOUNT_NATIVE_CURRENCY: { [chainId in ChainId]: CurrencyAmount } = {
+  [ChainId.RINKEBY]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.05', Token.getNative(ChainId.RINKEBY).decimals).toString(),
+    ChainId.RINKEBY
+  ),
+  [ChainId.MAINNET]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.1', Token.getNative(ChainId.MAINNET).decimals).toString(),
+    ChainId.MAINNET
+  ),
+  [ChainId.XDAI]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('1000', Token.getNative(ChainId.XDAI).decimals).toString(),
+    ChainId.XDAI
+  ),
+  [ChainId.SOKOL]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('1000', Token.getNative(ChainId.SOKOL).decimals).toString(),
+    ChainId.SOKOL
+  ),
+  [ChainId.ARBITRUM_TESTNET_V3]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.05', Token.getNative(ChainId.ARBITRUM_TESTNET_V3).decimals).toString(),
+    ChainId.ARBITRUM_TESTNET_V3
+  )
+}
+
+export class LiquidityMiningCampaign {
+  public readonly chainId: ChainId
+  public readonly address?: string
+  public readonly startsAt: BigintIsh
+  public readonly endsAt: BigintIsh
+  public readonly rewards: PricedTokenAmount[]
+  public readonly targetedPair: Pair
+  public readonly staked: PricedTokenAmount
+  public readonly duration: BigintIsh
+  public readonly locked: boolean
+
+  constructor(
+    startsAt: BigintIsh,
+    endsAt: BigintIsh,
+    targetedPair: Pair,
+    rewards: PricedTokenAmount[],
+    staked: PricedTokenAmount,
+    locked: boolean,
+    address?: string
+  ) {
+    invariant(JSBI.lessThan(parseBigintIsh(startsAt), parseBigintIsh(endsAt)), 'INCINSISTENT_DATES')
```

Comment:
Inconsistent spelling :P

Reply:
Ahhhh missed this. Fixed here: a484ec41fa30f78250de1a8c0f5c127d265b3d23